libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts

```diff
@@ -150,4 +151,26 @@ export class SegmentedControlComponent<TItem extends SegmentItem = SegmentItem>
       });
     }
   }
+
+  focusNativeButton(event: UIEvent) {
+    (event.target as HTMLIonSegmentButtonElement)?.setFocus();
+  }
+
+  getTabIndex(item: TItem, index: number) {
+    const segmentElementHasFocus = this.ionSegmentElement.nativeElement.matches(':focus-within');
+    // When focused prevent tab stop from inner native button to outer ion-segment-button:
+    if (segmentElementHasFocus) return -1;
+    // Allow tab stop on selected item:
+    if (item.id === this.value?.id) return null;
+    // Allow tab stop on first item if no value is set:
+    if (!this.value && index === 0) return null;
+    // Prevent tab stop on all other items:
+    return -1;
+  }
+
+  @HostListener('focusin', ['$event'])
```

Comment:
Good point 👌🏻 

In practice the method is only called on init and on focusin/out (i.e. tab into component, navigatin with arrow keys and selecting an item) so as far as I can see it would result in the exact same number of methods calls anyhoo 🤷🏻

Reply:
I think it might resolve in more method calls than you're suggesting, since angular doesn't know which value the function will return and thereby it has to rerun it a lot of times, until it has convinced itself that the value will not change. 
**Example with segmented control having 3 segmented buttons:** 
When angular reaches the first button that needs to be rendered it calls the getTabIndex() function and gets a value for the attr.tabindex for the first button. Then it reaches the second button and calls the same function. However now angular doesn't know if second time the function was called, changed some things that could result in a different value for the first button - and now it has to rerun that again for the first button. It will keep rerunning this function for both buttons until it is convinced that they won't change. And then angular reaches the third button - calls the function again but now it needs to check with the first button and second button again.

All in all a lot of change detection cycles run - but again even 100cycles doesn't increase performance so I dont think we should do anything with this.

Reply:
@Fuzzy3 as we talked about at demo last week this is not the case 🤓